### ISSUE TITLE
'docker info' over swarm a bit hard to parse

### ISSUE THREAD

**Issue by orkaa** [docker/swarm#1137]
I was working on my python app using swarm and docker-py and I tried to get some info about nodes in swarm.

Docker-py just proxies what it gets from the json api:

```
http://localhost:2376/v1.19/info
{ u'Containers': 5,
  u'Debug': False,
  u'DriverStatus': [ [u'\x08Strategy', u'spread'],
                     [ u'\x08Filters',
                       u'affinity, health, constraint, port, dependency'],
                     [u'\x08Nodes', u'1'],
                     [u'orkaa-dev', u'172.17.42.1:2375'],
                     [u' \u2514 Containers', u'5'],
                     [u' \u2514 Reserved CPUs', u'0 / 2'],
                     [u' \u2514 Reserved Memory', u'0 B / 2.061 GiB']],
  u'NEventsListener': 0}
```

Docker cli renders the above to:

```
$ docker -H localhost:2376 info
Containers: 5
Images: 0
Strategy: spread
Filters: affinity, health, constraint, port, dependency
Nodes: 1
 orkaa-dev: 172.17.42.1:2375
  └ Containers: 5
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 2.061 GiB
CPUs: 0
Total Memory: 0 B
```

In the console it looks fine, but if you try to parse it straight from the api, it isn't very helpful. It's basically a non-json tree structure inside json.

Here's an example of the code that produces that output: https://github.com/docker/swarm/blob/v0.3.0/cluster/swarm/cluster.go#L532

Do you have any ideas for making this more api friendly? If yes, I can probably help coding it.

**Comment by chanwit** [comment]
It's been like this because Swarm adds extra information atop Docker info. I am quite sure this will be improved in the very near future.

We have an early discussion about these enhanced APIs in #735.

**Comment by orkaa** [comment]
Great news @chanwit.

I'm closing this ticket because it's a duplicate of #735
